### PR TITLE
[DO NOT MERGE] Akka persistence testkit experimenting. #15571

### PR DESCRIPTION
**Pull request by manonthegithub**
Not to loose the branch I decided to create PR. Here are some experiments on persistence testkit, please don't judge severely, it is raw and demonstrates only main idea. Probably this can be useful for persistence testkit implementers. The main question is about approach. What do you think guys, is it ok using shared mutable state? or probably assertion methods should be implemented via actor communication and ask pattern?
refs #15571

### CODE REVIEW

**Comment by zaharidichev** [comment]
I was looking at that and intend to pick it up and frankly speaking it looks like an okey approach to me. I also assume that journal should be cleared on every test. Alternative is to provide a separate actor system with a separate fresh journal for every test.

**Comment by manonthegithub** [comment]
@zaharidichev Currently, I do not have much time, so feel free to pick it up. But if you won't have enough time, I can catch it up finally.

**Comment by zaharidichev** [comment]
@manonthegithub Cool. I will pick it up as I have some spare time and we'll see where we go from there ok ?

**Comment by manonthegithub** [comment]
@zaharidichev Great!

File: akka-persistence-testkit/src/main/scala/akka/persistence/testkit/PersistenceTestKitImpl.scala [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+import scala.util.{ Failure, Success, Try }
+
+trait PersistenceTestKit extends TestKitBase with PersistentTestKitOps {
+
+  override implicit lazy val system = {
+    //todo probably implement method for setting plugin in Persistence for testing purposes
+    ActorSystem(
+      s"persistence-testkit-${UUID.randomUUID()}",
+      PersistenceTestKitPlugin.PersitenceTestkitPluginConfig
+        .withFallback(ConfigFactory.defaultApplication()))
```
Comment (patriknw) [inline akka-persistence-testkit/src/main/scala/akka/persistence/testkit/PersistenceTestKitImpl.scala]: `defaultApplication` should not be mandatory, it must be passed in from concrete test.

However, might be better to not extend TestKit, and instead provide this as a utility than something that is mixed in to the test?

Comment (patriknw) [reply akka-persistence-testkit/src/main/scala/akka/persistence/testkit/PersistenceTestKitImpl.scala]: that would also make it safer to add more things later, without risking name clash with user's test

**Review by patriknw — COMMENTED** [review]
I think this is in the right direction, a few comments...

**Comment by johanandren** [comment]
This seems to have gone into hiatus, can we close the PR for now and re-open when there is renewed work on it (or start a new PR at that time instead perhaps)?

**Comment by manonthegithub** [comment]
I’ve taken a break on it, but plan to return to it soon. This can live in my branch until I get to the point when it can be reviewed, so I think you can close it. It was originally created for someone who would pick it up and complete.

**Comment by johanandren** [comment]
Ok, thanks for verifying (and for the work on it)!

**Pull request closed without merging**